keps/sig-node/4191-split-image-filesystem/README.md

```diff
@@ -0,0 +1,954 @@
+
+## Motivation
+
+Kubelet has two distinct filesystems: Node and Image. In typical deployments, users deploy Kubernetes where both the Node and Image filesystems are on the same disk. There are some requests to separate the storage into separate disks.  The most common request is to separate the writable layer from the read-only layer. Kubelet and Container data would be stored on the same disk while images would have their own disk.  This could be beneficial because images occupy a lot of disk space while the writeable layer is typically smaller.
+
+Container IO can impact Kubelet and adding the ability for more disks could increase performance of Kubelet.
+
+However, it is not possible to separate the image layers and container writable layers on different disks.
+
+In the current implementation of separate disks, containers and images must be stored on the same disk.  So garbage collection, in case of node pressure (really image disk pressure) would GC images/containers on the image filesystem.
+
+If one separates writable layer (containers) from readable layer (images), then garbage collection and statistics must account for this separation.  Today this could potentially break Kubelet if the container runtime configures storage in this way.
+
+One downside of the separate disk is that pod data can be written in multiple locations.  The writeable layer of a container would go on the image filesystem and volume storage would go to the root fs.  There is another request to separate the root and the image filesystem to be writeable and read-only respective.  This means that pod data can be written on one disk while the other disk can be read-only.  Separting the writeable layer and the read-only layer will achieve this.
+
+### Goals
+
+- Kubelet should still work if images/containers are separated into different disks.
+  - Support writable layer being on same disk as Kubelet
+  - Images can be on the separate filesystem.
+
+### Possible Extensions in Post Alpha
+
+Kubelet, Images and Containers on all separate disks.
+
+This case is possible with this implementation as containerfs will be set up to read file statistics from a separate filesystem.  However, this is not in scope for Alpha.  
+If there is interest in this, this KEP could be extended to support this use case. Main areas to add would be testing.
+
+### Non-Goals
+
+- Multiple nodes can not share the same filesystem.
+- Separating kubelet data into different filesystems.
+- Multiple image and/or container filesystems
+  - This KEP will start support for this but more work needs to be done to investigate CAdvisor/CRIStats/Eviction to support this.
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+As a user, I would like to have my node configured so that I have a writeable filesystem and a readable filesystem.  Kubelet will write volume and writeable layers to the writeable filesystem while images will be written to a read-only filesystem.
```

Comment:
nit: Typically the container runtime is responsible for managing the layers.

Reply:
updated.